validator/src/server/cli.rs

```diff
@@ -0,0 +1,226 @@
+                .long("verbose")
+                .multiple(true)
+                .help("enable more verbose output to stderr"),
+        )
+        .arg(
+            Arg::with_name("scheduler")
+                .long("scheduler")
+                .takes_value(true)
+                .possible_values(&["serial", "parallel"])
+                .help("set scheduler type: serial or parallel"),
+        )
+        .arg(
+            Arg::with_name("network_auth")
+                .long("network-auth")
+                .takes_value(true)
+                .possible_values(&["trust", "challenge"])
+                .help(
+                    "identify type of authorization required to join validator \
+                     network.",
+                ),
+        )
+        .arg(
+            Arg::with_name("opentsdb_url")
+                .long("opentsdb-url")
+                .takes_value(true)
+                .help(
+                    "specify host and port for Open TSDB database used for \
+                     metrics",
+                ),
+        )
+        .arg(
+            Arg::with_name("opentsdb_db")
+                .long("opentsdb-db")
+                .takes_value(true)
+                .help("specify name of database used for storing metrics"),
+        )
+        .arg(
+            Arg::with_name("minimum_peer_connectivity")
+                .long("minimum-peer-connectivity")
+                .takes_value(true)
+                .help(
+                    "set the minimum number of peers required before stopping \
+                     peer search",
+                ),
+        )
+        .arg(
+            Arg::with_name("maximum_peer_connectivity")
+                .long("maximum-peer-connectivity")
+                .takes_value(true)
+                .help("set the maximum number of peers to accept"),
+        );
+
+    app.get_matches()
+}
+
+fn parse_roles<'a>(matches: &'a ArgMatches, py: &Python) -> Option<PyDict> {
+    match matches.value_of("network_auth") {
+        Some(network_auth) => {
+            let auth_dict = PyDict::new(*py);
+            auth_dict.set_item(*py, "network", network_auth).unwrap();
+            Some(auth_dict)
+        }
+        None => None,
+    }
+}
+
+fn parse_bindings<'a>(matches: &'a ArgMatches) -> (Option<&'a str>, Option<&'a str>) {
+    let mut bind_network = None;
+    let mut bind_component = None;
+
+    if let Some(bindings) = parse_multiple_args("bind", matches) {
+        for binding in bindings {
+            if binding.starts_with("network") {
+                bind_network = Some(binding.splitn(2, ':').skip(1).collect::<Vec<_>>()[0]);
+            }
+
+            if binding.starts_with("component") {
+                bind_component = Some(binding.splitn(2, ':').skip(1).collect::<Vec<_>>()[0]);
+            }
+        }
+    };
+
+    (bind_component, bind_network)
+}
+
+fn parse_comma_separated_args(name: &str, matches: &ArgMatches) -> Option<Vec<String>> {
```

Comment:
Can this also return a `Option<Vec<&'a str>>`, like the following function?

Reply:
We need to go from a list of possibly comma-separated args like `["a,b,c", "d", "e"]` to a list of separated args `["a", "b", "c", "d", "e"]`. In Python this was done by creating a new list and then splitting the args and appending them one by one. My idea was to join all the strings with a comma and then split on the comma. Unfortunately `join` (I think) returns `String`, so that's what the function has to return.

Is there a more elegant way to do it? I don't like the return type either, but I didn't want to spend a lot of time on it since the function gets called just twice.